tests/test_cases/test_cocotb/test_cocotb.py

```diff
@@ -1266,5 +1267,50 @@ def test_expect_exception_list(dut):
     raise MyException()
 
 
+@cocotb.coroutine
+def example():
+    yield Timer(10, 'ns')
+    raise ReturnValue(1)
+
+
+@cocotb.test()
+def test_timeout_func_fail(dut):
+    try:
+        yield cocotb.triggers.with_timeout(example(), timeout_time=1, timeout_unit='ns')
+    except cocotb.result.SimTimeoutError:
+        pass
+    else:
+        assert False, "Expected a Timeout"
+
+
+@cocotb.test()
+def test_timeout_func_pass(dut):
+    res = yield cocotb.triggers.with_timeout(example(), timeout_time=100, timeout_unit='ns')
+    assert res == 1
+
+
+@cocotb.test(expect_error=True, timeout_time=1, timeout_unit='ns')
+def test_timeout_testdec_fail(dut):
+    yield Timer(10, 'ns')
+
+
+@cocotb.test(timeout_time=100, timeout_unit='ns')
+def test_timeout_testdec_pass(dut):
+    yield Timer(10, 'ns')
+
+
+@cocotb.test(timeout_time=10, timeout_unit='ns')
+def test_timeout_testdec_simultaneous(dut):
+    try:
+        yield cocotb.triggers.with_timeout(Timer(1, 'ns'), timeout_time=1, timeout_unit='ns')
+    except cocotb.result.SimTimeoutError:
+        pass
+    else:
+        assert False, "Expected a Timeout"
+    # Whether this test fails or passes depends on the behavior of the
+    # scheduler and the implementation of the timeout function.
```

Comment:
Not only the scheduler, but the simulator too. You're right to claim this, we draw attention to it in the `First` docs here: https://external-builds.readthedocs.io/html/cocotb/1119/triggers.html#cocotb.triggers.First

(no action needed)